### ISSUE TITLE
Restic mount hangs in the background after mount/unmount

### ISSUE THREAD

**Issue by kobuki** [restic/restic#2298]
Output of `restic version`
--------------------------
restic 0.9.5 compiled with go1.12.4 on linux/amd64

How did you run restic exactly?
-------------------------------
```
# export RESTIC_PASSWORD=xxx
# restic -r /mnt/backup/restic/root mount /mnt/temp/ &
repository 2fc17b90 opened successfully, password is correct
Now serving the repository at /mnt/temp/
When finished, quit with Ctrl-c or umount the mountpoint.
```

What backend/server/service did you use to store the repository?
----------------------------------------------------------------

CIFS network mount or local ZFS dataset (both exhibit the issue)

Expected behavior
-----------------

The command mounts the repo at /mnt/temp and at unmount, exits. The repo is fine, no problems accessing snapshots under the mount, etc. Note that the command is run in the background using `&`.

Actual behavior
---------------

When I unmount the repo using `umount /mnt/temp`, it's succesfully unmounted, but restic doesn't quit, it stays as a hung process in the background.

Steps to reproduce the behavior
-------------------------------

See the command issued for mounting.

Do you have any idea what may have caused this?
-----------------------------------------------

Nope.

Do you have an idea how to solve the issue?
-------------------------------------------

Nope.

Did restic help you or made you happy in any way?
-------------------------------------------------

Don't know yet, as I've just started using it for daily workloads.

**Comment by nunoperalta** [comment]
I got this same problem. Had to kill -9 the restic mount process.
Also, "ls /mnt/restic" was also hanging forever.

**Comment by MichaelEischer** [comment]
Restic will properly stop once you pull it back into the foreground with `fg`.

I've took a closer look at what's happening: After unmounting, the fuse serve loop exits and restic starts to shut down. This triggers the cleanup hooks, in particular `restoreTerminal` which was added to fix #402. That hook restores the terminal settings to the configuration it had when restic was started. Problem is that a background process is by default paused (via the `SIGTTOU` signal) when trying to reconfigure the terminal. And that means that restic hangs until it's moved to the foreground again.

The hook only has an effect when restic was interrupted while it tried to read a password from the terminal. Or put the other way around: restic should only restore the terminal configuration when it was interrupted while reading a password from the terminal.

https://github.com/restic/restic/blob/5a7c27ddb62d86440e60764fab4725e520f78e54/cmd/restic/global.go#L163

In fact, this problem with running as a background job, affects every single restic command, not just `mount`.

I'll put together a pull request shortly.
